packages/cms-lib/lib/config/portal.js

```diff
@@ -0,0 +1,160 @@
+const { logger } = require('../../logger');
+const {
+  API_KEY_AUTH_METHOD,
+  OAUTH_AUTH_METHOD,
+  PERSONAL_ACCESS_KEY_AUTH_METHOD,
+} = require('../constants');
+const {
+  getAndLoadConfigIfNeeded,
+  setDefaultConfigPathIfUnset,
+  writeConfig,
+} = require('./file');
+const { getUpdatedApiKeyPortalConfig } = require('./apiKey');
+const { getUpdatedOauthPortalConfig } = require('./oauth2');
+const {
+  getUpdatedPersonalAccessKeyPortalConfig,
+} = require('./personalAccessKey');
+
+const getPortalConfig = portalId => {
+  const config = getAndLoadConfigIfNeeded();
+  return config.portals.find(portal => portal.portalId === portalId);
+};
+
+const getPortalId = nameOrId => {
+  const config = getAndLoadConfigIfNeeded();
+  let name;
+  let portalId;
+  let portal;
+  if (!nameOrId) {
+    if (config && config.defaultPortal) {
+      name = config.defaultPortal;
+    }
+  } else {
+    if (typeof nameOrId === 'number') {
+      portalId = nameOrId;
+    } else if (/^\d+$/.test(nameOrId)) {
+      portalId = parseInt(nameOrId, 10);
+    } else {
+      name = nameOrId;
+    }
+  }
+
+  if (name) {
+    portal = config.portals.find(p => p.name === name);
+  } else if (portalId) {
+    portal = config.portals.find(p => p.portalId === portalId);
+  }
+
+  if (portal) {
+    return portal.portalId;
+  }
+
+  return null;
+};
+
+const getPortalName = portalId => {
+  return getPortalConfig(portalId).name;
+};
+
+/**
+ * Method to update an existing portal config or create a new one
+ * @param {object} configOptions An object containing properties to update in the portalConfig
+ * @param {number} configOptions.portalId Portal ID to add/make updates to
+ * @param {string} configOptions.authType Type of authentication used for this portalConfig
+ * @param {string} configOptions.env Environment that this portal is located in(QA/PROD)
+ * @param {string} configOptions.name Unique name used to reference this portalConfig
+ * @param {string} configOptions.apiKey API key used in authType: apikey
+ * @param {string} configOptions.defaultMode Default mode for uploads(draft or publish)
+ * @param {string} configOptions.personalAccessKey Personal Access Key used in authType: personalaccesskey
+ * @param {object} configOptions.auth Auth object used in oauth2 and personalaccesskey authTypes
+ * @param {string} configOptions.auth.clientId Client ID used for oauth2
+ * @param {string} configOptions.auth.clientSecret Client Secret used for oauth2
+ * @param {array} configOptions.auth.scopes Scopes that are allowed access with auth
+ * @param {object} configOptions.auth.tokenInfo Token Info used for oauth2 and personalaccesskey authTypes
+ * @param {string} configOptions.auth.tokenInfo.accessToken Access token used for auth
+ * @param {string} configOptions.auth.tokenInfo.expiresAt Date ISO of accessToken expiration
+ */
+const updatePortalConfig = configOptions => {
```

Comment:
It looks like the shape expected for `configOptions` has changed, but all of the calls to `updatePortalConfig` have not been updated.

Reply:
Here is the prior shape and changes made
```
const {	
    portalId,	
    authType,	
    environment,	// Renamed to env
    clientId,	        // Moved into auth prop
    clientSecret,	// Moved into auth prop
    scopes,	        // Moved into auth prop
    tokenInfo,	        // Moved into auth prop
    defaultMode,	
    name,	
    apiKey,	
    personalAccessKey,	
  } = configOptions;
```

For the items that were moved into the `auth` prop, they were coming from the `OAuth2Manager.toObj` and `personalAccessKey.updateConfigWithPersonalAccessKey` methods, which I updated with proper nesting of these props under `auth`.

I think this is a good change because it makes the shape consistent instead of passing all top level prop data to updatePortalConfig and then relying on on this method to properly nest 3 different shapes of data.

Reply:
When I was reading through the code there were calls that weren’t updated. I think that they were related to refreshing access tokens for both access keys and standard OAuth.

Reply:
AH sorry, you are correct. I missed those two -- will fix.